### PR TITLE
feat: auto update browserslist DB

### PR DESCRIPTION
**Pull request by abdullahwaheed**
### Ticket
[Automate browserslist DB update ](https://github.com/openedx/frontend-wg/issues/101)

### What has changed
Added a Github workflow that would run periodically(weekly), update browserslist DB, [create a PR](https://github.com/marketplace/actions/create-pull-request) and [merge](https://github.com/peter-evans/enable-pull-request-automerge) it automatically.
Used a shared [workflow](https://github.com/openedx/.github/blob/master/.github/workflows/update-browserslist-db.yml) for this

### CODE REVIEW

File: .github/workflows/update-browserlist.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        node-version: ${{ matrix.node }}
+    - run: npm install -g npm@${{ matrix.npm }}
+    - run: make requirements
+
+    - name: Update dependencies
+      run: npx browserslist@latest --update-db
+
+    - name: setup testeng-ci
+      run: |
+        git clone https://github.com/edx/testeng-ci.git
```
Comment (arbrandes) [inline .github/workflows/update-browserlist.yml:28]: It looks like this PR creator is something that would be widely useful to the Open edX community.  (I mean, it's already going into all MFEs.  :)  Would it be possible for us to get in touch with the maintainer of the `testeng-ci` repository, and then to consider copying the relevant bits into the openedx org?

**Review by arbrandes — COMMENTED** [review]
My concern is that using testeng-ci breaks Rule #2 [here](https://discuss.openedx.org/t/interim-guidance-on-the-openedx-github-organization/8518).

"Repos must be added to the openedx org before they become critical dependencies of any other repository in the openedx organization."

Not sure how critical I'd classify the automatic browserlist update, but as I note below, my gut feeling is that we'd want the (a?) PR creator in the openedx org itself.

(We can discuss this in the next Frontend Working Group meeting, too.)

File: .github/workflows/update-browserlist.yml (hunk 1 above)
Comment (abdullahwaheed) [reply .github/workflows/update-browserlist.yml:28]: I am not sure about this, @jmbowman can comment on this

**Comment by arbrandes** [comment]
Alright, I'm up for merging this.  @abdullahwaheed, mind rebasing?

**Comment by abdullahwaheed** [comment]
> Alright, I'm up for merging this. @abdullahwaheed, mind rebasing?

@arbrandes we are moving towards using this [create-pull-request](https://github.com/marketplace/actions/create-pull-request) package, so let me update this PR accordingly and then we can merge.

Please refer to [this](https://edx-internal.slack.com/archives/C03K11R544U/p1667510391024659) thread for details

**Review by arbrandes — APPROVED** [review]
Approved, and repo auto-merge set.

**Pull request merged**